exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceRuntime.java

```diff
@@ -75,7 +69,7 @@
   /**
    * The active services indexed by their name. It is stored in a sorted map that offers
    * the same iteration order on all nodes with the same services, which is required
-   * for correct operation of beforeCommit and {@link #getStateHashes(Snapshot)}.
+   * for correct operation of state hash aggregation.
```

Comment:
Should I place a link to Schema here?

Reply:
Actually, it is no longer required to be a sorted map because no order-sensitive iteration occurs in the runtime (@slowli — will there be in a near future? Shall we replace with a HashMap?) — afterCommit is not such.

--- 

Even if there are none order-sensitive operations *but* we log all services (currently I don't see such logs) it might be reasonable to keep to ease log navigation from different nodes.

Reply:
> will there be in a near future?

Not that I'm aware of. The general idea is that operations pertaining to services (e.g., `before_transactions` / `after_transactions` hooks) are ordered *by the core* and called for each service separately.

Reply:
That's good, thanks.